pkg/devfile/validate/events.go

```diff
@@ -0,0 +1,108 @@
+package validate
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/devfile/api/pkg/apis/workspaces/v1alpha1"
+	"github.com/devfile/parser/pkg/devfile/parser/data"
+	"k8s.io/klog"
+)
+
+// validateEvents validates all the devfile events
+func validateEvents(data data.DevfileData) error {
+
+	eventErrors := ""
+	events := data.GetEvents()
+
+	switch {
+	case len(events.PreStart) > 0:
+		klog.V(4).Info("Validating preStart events")
+		if preStartErr := isEventValid(data, events.PreStart, "preStart"); preStartErr != nil {
+			eventErrors += fmt.Sprintf("\n%s", preStartErr.Error())
```

Comment:
It might be cleaner to have
```
var eventErrors []string
//...
eventErrors = append(eventErrors, preStartErr.Error())
//...
return fmt.Errorf("devfile events validation error:\n%s", strings.Join(eventErrors, "\n"))
```